In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.patches import Rectangle


## DATASET 1: CALIDAD DEL AGUA

In [18]:
pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/waterquality.csv')



aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
1          2.32   21.16     0.01    3.31    0.002        5.28      0.68   
2          1.01   14.02     0.04    0.58    0.008        4.24      0.53   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
1       0.66      0.90      0.65  ...  0.100      2.01      1.93    0.003   
2       0.02      0.99      0.05  ...  0.078     14.16      1.11    0.006   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
1           32.26    3.21      0.08    0.27     0.05        1  
2           50.28    7.07      0.07    0.44     0.01        0  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7999 rows x 21 columns]

In [17]:
df = pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/waterquality.csv')


(7999, 21)

In [21]:
df.head(20)


aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  copper  \
0        1.65    9.08    0.040    2.85    0.007        0.35      0.83    0.17   
1        2.32   21.16    0.010    3.31    0.002        5.28      0.68    0.66   
2        1.01   14.02    0.040    0.58    0.008        4.24      0.53    0.02   
3        1.36   11.33    0.040    2.96    0.001        7.23      0.03    1.66   
4        0.92   24.33    0.030    0.20    0.006        2.67      0.69    0.57   
5        0.94   14.47    0.030    2.88    0.003        0.80      0.43    1.38   
6        2.36     5.6    0.010    1.35    0.004        1.28      0.62    1.88   
7        3.93   19.87    0.040    0.66    0.001        6.22      0.10    1.86   
8        0.60   24.58    0.010    0.71    0.005        3.14      0.77    1.45   
9        0.22   16.76    0.020    1.37    0.007        6.40      0.49    0.82   
10       3.27     3.6    0.001    2.69    0.005        5.75      0.15    0.60   
11       1.35   21.96    0.040    0.84    0.002        0.10      0.76    0.17   
12       1.88   19.26    0.020    2.78    0.008        0.05      0.42    1.00   
13       4.93   23.98    0.040    3.05    0.008        0.70      0.51    1.35   
14       2.89   18.82    0.050    3.77    0.008        5.99      0.54    0.79   
15       0.61    2.41    0.030    0.59    0.002        1.94      0.77    1.54   
16       3.47   15.84    0.020    0.06    0.001        5.29      0.47    1.08   
17       2.11   17.03    0.020    0.88    0.009        7.78      0.88    1.15   
18       4.88   26.94    0.020    0.36    0.001        1.21      0.68    0.71   
19       4.12   17.99    0.020    3.43    0.006        0.01      0.41    1.82   

    flouride  bacteria  ...   lead  nitrates  nitrites  mercury  perchlorate  \
0       0.05      0.20  ...  0.054     16.08      1.13    0.007        37.75   
1       0.90      0.65  ...  0.100      2.01      1.93    0.003        32.26   
2       0.99      0.05  ...  0.078     14.16      1.11    0.006        50.28   
3       1.08      0.71  ...  0.016      1.41      1.29    0.004         9.12   
4       0.61      0.13  ...  0.117      6.74      1.11    0.003        16.90   
5       0.11      0.67  ...  0.135      9.75      1.89    0.006        27.17   
6       0.33      0.13  ...  0.021     18.60      1.78    0.007        45.34   
7       0.86      0.16  ...  0.197     13.65      1.81    0.001        53.35   
8       0.98      0.35  ...  0.167     14.66      1.84    0.004        23.43   
9       1.24      0.83  ...  0.109      4.79      1.46    0.010        30.42   
10      1.29      0.04  ...  0.145      8.47      1.25    0.006        55.40   
11      0.58      0.52  ...  0.011     18.40      1.49    0.009        21.52   
12      0.09      0.91  ...  0.103      4.37      1.95    0.006        22.12   
13      1.07      0.70  ...  0.101      1.16      1.11    0.008        26.80   
14      0.54      0.20  ...  0.126     17.56      1.82    0.009        17.54   
15      0.62      0.23  ...  0.017      1.99      1.08    0.007        11.16   
16      1.43      0.89  ...  0.080      1.91      1.20    0.008         0.18   
17      0.34      0.85  ...  0.065     17.86      1.53    0.003        19.40   
18      0.99      0.75  ...  0.071      0.31      1.22    0.002        56.70   
19      0.22      0.99  ...  0.108      8.06      1.76    0.005        24.29   

    radium  selenium  silver  uranium  is_safe  
0     6.78      0.08    0.34     0.02        1  
1     3.21      0.08    0.27     0.05        1  
2     7.07      0.07    0.44     0.01        0  
3     1.72      0.02    0.45     0.05        1  
4     2.41      0.02    0.06     0.02        1  
5     5.42      0.08    0.19     0.02        1  
6     2.84      0.10    0.24     0.08        0  
7     7.24      0.08    0.08     0.07        0  
8     4.99      0.08    0.25     0.08        1  
9     0.08      0.03    0.31     0.01        1  
10    7.80      0.05    0.33     0.06        0  
11    1.30      0.08    0.48     

In [22]:
df.tail(20)


aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
7979       0.03   11.73     0.01    0.50     0.10        0.30      0.09   
7980       0.02    0.11     0.06    0.96     0.05        0.34      0.04   
7981       0.09    0.15     0.09    0.14     0.05        0.12      0.07   
7982       0.02    4.34     0.09    0.67     0.07        0.03      0.04   
7983       0.10    0.83     0.08    2.85     0.07        0.46      0.06   
7984       0.00    10.9     0.07    0.28     0.09        0.29      0.06   
7985       0.01    0.21     0.04    0.54     0.10        0.10      0.01   
7986       0.03   20.17     0.08    0.41     0.09        0.08      0.03   
7987       0.04    3.04     0.08    0.57     0.09        0.04      0.07   
7988       0.07    0.17     0.08    0.57     0.01        0.01      0.05   
7989       0.01   27.73     0.06    0.77     0.09        0.20      0.03   
7990       0.00    3.11     0.07    1.24     0.07        0.06      0.03   
7991       0.03    8.94     0.02    1.25     0.08        0.22      0.02   
7992       0.04   10.84     0.06    0.49     0.04        0.08      0.03   
7993       0.02    5.24     0.05    0.77     0.02        0.57      0.04   
7994       0.05    7.78     0.00    1.95     0.04        0.10      0.03   
7995       0.05   24.22     0.02    0.59     0.01        0.45      0.02   
7996       0.09    6.85     0.00    0.61     0.03        0.05      0.05   
7997       0.01      10     0.01    2.00     0.00        2.00      0.00   
7998       0.04    6.85     0.01    0.70     0.03        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
7979    0.06      1.43      0.16  ...  0.075     14.67       1.0    0.002   
7980    0.04      0.21      0.00  ...  0.111      4.74       1.0    0.001   
7981    0.07      1.06      0.00  ...  0.061      1.71       1.0    0.000   
7982    0.08      0.53      0.00  ...  0.159     15.32       1.0    0.006   
7983    0.01      0.11      0.00  ...  0.058      8.99       1.0    0.001   
7984    0.07      0.81      0.00  ...  0.066      5.51       1.0    0.000   
7985    0.01      1.30      0.00  ...  0.154     15.79       1.0    0.007   
7986    0.03      1.16      0.00  ...  0.114      0.83       1.0    0.005   
7987    0.06      0.96      0.00  ...  0.138      6.15       1.0    0.000   
7988    0.03      0.44      0.00  ...  0.047     14.61       1.0    0.002   
7989    0.00      0.58      0.00  ...  0.112     11.74       1.0    0.007   
7990    0.04      0.74      0.00  ...  0.016     15.15       1.0    0.005   
7991    0.05      0.71      0.00  ...  0.112     11.97       1.0    0.001   
7992    0.08      0.49      0.00  ...  0.151     14.44       1.0    0.003   
7993    0.03      0.15      0.00  ...  0.085     19.41       1.0    0.001   
7994    0.03      1.37      0.00  ...  0.197     14.29       1.0    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27       1.0    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92       1.0    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00       0.0    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92       1.0    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
7979         8.44    3.34      0.00    0.09     0.02        0  
7980        10.60    7.92      0.09    0.02     0.04        0  
7981         0.24    5.22      0.01    0.03     0.06        0  
7982         4.64    0.84      0.02    0.03     0.03        1  
7983         0.82    7.51      0.00    0.05     0.09        0  
7984         8.74    5.73      0.07    0.02     0.02        0  
7985         2.69    5.69      0.09    0.05     0.04        0  
7986         3.15    4.16      0.05    0.00     0.06        1  
7987         0.99    0.20      0.01    0.08     0.06        1  
7988         1.50    3.95      0.05    0.09     0.00        1  
7989         7.58    7.21      0.07    0.04     0.09        0  
7990         0.46    6.24      0.03    0.09 

In [23]:
df.sample(20)


aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
6267       0.07   10.12    0.060    1.13    0.030        0.37      0.08   
1803       4.41    29.4    0.490    4.25    0.010        1.65      0.88   
2722       0.03   22.79    0.020    0.45    0.005        7.08      0.62   
1313       0.54   28.08    0.270    1.01    0.110        2.22      0.65   
293        3.89    1.05    0.010    2.28    0.008        1.26      0.04   
5673       0.06    6.79    0.050    0.56    0.060        0.04      0.04   
7506       0.09    4.21    0.060    0.50    0.090        0.07      0.06   
3841       0.05   29.74    0.470    0.79    0.050        5.30      0.08   
5121       0.09   29.41    0.090    1.83    0.070        0.19      0.08   
4937       0.10    9.85    0.030    0.22    0.070        0.16      0.08   
3023       0.09   20.54    0.960    2.88    0.020        7.65      0.61   
3964       0.06   23.44    0.720    3.19    0.130        7.12      0.40   
7728       0.00    2.88    0.020    2.31    0.030        0.29      0.03   
2542       0.01   26.55    0.001    3.08    0.007        7.87      0.37   
3148       0.01    9.24    0.290    2.14    0.060        7.61      0.19   
5891       0.10      17    0.060    1.23    0.070        0.03      0.04   
1911       3.76    6.42    0.570    0.48    0.050        0.40      0.47   
4286       0.03   23.91    0.030    2.55    0.030        0.07      0.01   
1297       0.24       5    0.310    2.56    0.100        7.89      0.05   
3255       0.07    7.79    0.870    2.76    0.080        6.74      0.06   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
6267    1.07      1.35      0.39  ...  0.113     14.25      1.24    0.003   
1803    1.19      0.38      0.84  ...  0.064      8.12      1.16    0.005   
2722    0.61      1.46      0.84  ...  0.136     16.56      1.62    0.007   
1313    0.62      0.44      0.00  ...  0.136     12.40      1.34    0.004   
293     0.14      0.77      0.24  ...  0.062     13.59      1.98    0.005   
5673    1.47      0.37      0.74  ...  0.130     17.77      0.46    0.006   
7506    1.99      0.81      0.00  ...  0.098      4.75      1.66    0.003   
3841    0.09      0.69      0.77  ...  0.110      0.85      1.33    0.003   
5121    1.16      1.06      0.00  ...  0.120      7.09      0.26    0.001   
4937    0.01      0.90      0.00  ...  0.014     17.66      0.99    0.007   
3023    1.23      1.44      0.56  ...  0.012     10.12      2.07    0.004   
3964    0.03      0.52      0.49  ...  0.057     13.89      1.51    0.005   
7728    0.09      0.91      0.00  ...  0.191      1.81      1.42    0.005   
2542    1.75      0.19      0.90  ...  0.045      4.35      1.32    0.007   
3148    1.22      1.06      0.27  ...  0.103      3.71      1.89    0.007   
5891    0.87      0.30      0.76  ...  0.022      7.12      0.07    0.009   
1911    1.58      0.43      0.98  ...  0.187      9.61      2.68    0.001   
4286    0.04      0.07      0.00  ...  0.042     16.71      0.24    0.009   
1297    0.15      1.19      0.00  ...  0.164     17.72      1.56    0.001   
3255    0.67      0.63      0.48  ...  0.007      9.53      1.32    0.002   

      perchlorate  radium  selenium  silver  uranium  is_safe  
6267         0.38    4.33      0.06    0.07     0.04        0  
1803         7.10    3.72      0.07    0.42     0.06        0  
2722        37.96    5.67      0.06    0.15     0.08        0  
1313        33.81    2.12      0.09    0.37     0.02        0  
293         51.76    0.99      0.08    0.49     0.08        1  
5673         0.23    0.97      0.06    0.00     0.03        0  
7506         4.38    0.06      0.10    0.06     0.02        0  
3841        33.74    3.34      0.06    0.48     0.04        0  
5121         2.17    3.50      0.04    0.04     0.04        0  
4937         6.04    2.78      0.10    0.04     0.04        0  
3023        56.40    3.22      0.07    0.18     0.08        0  
3964        30.57    3.10      0.07    0.31 

#### VALORES FALTANTES DEL DATASET

In [24]:
df.dropna()

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
1          2.32   21.16     0.01    3.31    0.002        5.28      0.68   
2          1.01   14.02     0.04    0.58    0.008        4.24      0.53   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
1       0.66      0.90      0.65  ...  0.100      2.01      1.93    0.003   
2       0.02      0.99      0.05  ...  0.078     14.16      1.11    0.006   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
1           32.26    3.21      0.08    0.27     0.05        1  
2           50.28    7.07      0.07    0.44     0.01        0  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7999 rows x 21 columns]

In [25]:
df.fillna(0)

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
1          2.32   21.16     0.01    3.31    0.002        5.28      0.68   
2          1.01   14.02     0.04    0.58    0.008        4.24      0.53   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
1       0.66      0.90      0.65  ...  0.100      2.01      1.93    0.003   
2       0.02      0.99      0.05  ...  0.078     14.16      1.11    0.006   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
1           32.26    3.21      0.08    0.27     0.05        1  
2           50.28    7.07      0.07    0.44     0.01        0  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7999 rows x 21 columns]

### SOBRE EL DATASET

CONTEXTO ANALITICO

    El Dataset con el que se trabajará a continuación corresponde a un conjunto de datos referidos al contenido de minerales del agua en un pueblo llamado Dioncaster, el cual se ubica en el extremo occidental del condado de Sabbatown.
    El archivo .csv utilizado fue descargado de la página kaggle.com.

CONTEXTO COMERCIAL

    El objetivo del presente trabajo es analizar como impacta el contenido de determinados minerales en la calidad de agua, la cual es utilizada tanto para riego de cultivos, como para consumo animal. Por lo cual se pretende discriminar en cuanto a la aptitud del agua, en base al fin con el que se la vaya a utilizar.

DESCRIPCIÓN  

Todos los atributos del dataset listados a continuación están expresados por litro:
- aluminium: aluminio
- arsenic: arsénico
- barium: bario
- cadmium: cadmio
- chloramine: cloramina
- chromium: cromo
- cooper: cobre
- fluoride: fluoruro
- bacteria: bacterias
- viruses: virus
- lead: plomo
- nitrates: nitratos
- nitrites: nitritos
- mercury: mercurio
- perchlorate: perclorato
- radium: radio
- selenium: selenio
- silver: plata
- uranium: uranio
- is_safe: es seguro para consumo



El dataset seleccionado cuenta con 7999 filas y 21 columnas

In [15]:
display(df.shape)


(7999, 21)

El dataset cuenta con los siguientes nombres en sus columnas

In [11]:
df.columns

Index(['aluminium', 'ammonia', 'arsenic', 'barium', 'cadmium', 'chloramine',
       'chromium', 'copper', 'flouride', 'bacteria', 'viruses', 'lead',
       'nitrates', 'nitrites', 'mercury', 'perchlorate', 'radium', 'selenium',
       'silver', 'uranium', 'is_safe'],
      dtype='object')

El dataset con los siguientes tipos de datos en sus columnas

In [13]:
df.dtypes

aluminium      float64
ammonia         object
arsenic        float64
barium         float64
cadmium        float64
chloramine     float64
chromium       float64
copper         float64
flouride       float64
bacteria       float64
viruses        float64
lead           float64
nitrates       float64
nitrites       float64
mercury        float64
perchlorate    float64
radium         float64
selenium       float64
silver         float64
uranium        float64
is_safe         object
dtype: object

La información proporcionada es la siguiente:

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7999 entries, 0 to 7998
Data columns (total 21 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   aluminium    7999 non-null   float64
 1   ammonia      7999 non-null   object 
 2   arsenic      7999 non-null   float64
 3   barium       7999 non-null   float64
 4   cadmium      7999 non-null   float64
 5   chloramine   7999 non-null   float64
 6   chromium     7999 non-null   float64
 7   copper       7999 non-null   float64
 8   flouride     7999 non-null   float64
 9   bacteria     7999 non-null   float64
 10  viruses      7999 non-null   float64
 11  lead         7999 non-null   float64
 12  nitrates     7999 non-null   float64
 13  nitrites     7999 non-null   float64
 14  mercury      7999 non-null   float64
 15  perchlorate  7999 non-null   float64
 16  radium       7999 non-null   float64
 17  selenium     7999 non-null   float64
 18  silver       7999 non-null   float64
 19  uraniu

### Las variables que se busca predecir es la llamada "is_safe" a=is_safe 
### Es decir si el agua es apta para consumo animal.


#### Las variables consideradas relevantes para predecir la aptitud en consumo animal son z1=arsenic z2=fluorides z3=nitrates

#### z1=arsenic y z2=fluorides se ha seleccionado ya que el As y el F por si mismos pueden definir la ineptitud del agua para ser consumida. SOn elementos altamente tóxicos para animales y para el hombre, por lo que independientemente de la composición salina del agua en bebida, altos niveles de uno u otro limitan su uso.
#### z3=nitrates se ha elegido ya que los nitratos se consideran peligrosos a partir de 200 ppm y son indicadores de contaminación orgánica, también se ha encontrado que están muy relacionados a la fertilización irracional con urea.

In [27]:
print (type(df['is_safe']),type(df[['is_safe']]))
display(df['is_safe'].head(20))
display(df[['is_safe']].head(20))


<class 'pandas.core.series.Series'> <class 'pandas.core.frame.DataFrame'>


0     1
1     1
2     0
3     1
4     1
5     1
6     0
7     0
8     1
9     1
10    0
11    1
12    1
13    1
14    1
15    1
16    1
17    1
18    0
19    1
Name: is_safe, dtype: object

is_safe
0        1
1        1
2        0
3        1
4        1
5        1
6        0
7        0
8        1
9        1
10       0
11       1
12       1
13       1
14       1
15       1
16       1
17       1
18       0
19       1

## DATASET 2: CALIDAD DEL AGUA